### PR TITLE
Added XSPEC contents, XSPEC walkthrough, and XMM-SAS setup

### PR DESCRIPTION
**Pull request by phajy**
Added table of contents to XSPEC page and a link to the walk through tutorial. Added XMM-SAS installation instructions. Also fixed a few spelling mistakes.

### CODE REVIEW

File: guides/sas-setup.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+```
+wget http://sasdev-xmm.esac.esa.int/pub/sas/20.0.0/macOS/11.6/sas_20.0.0-macOS-11.6.tgz
+tar zxf sas_20.0.0-macOS-11.6.tgz
+export SAS_PERL=/usr/bin/perl
+./install.sh
+source /Users/my_user/Software/xmm-sas/xmmsas_20211130_0941/setsas.sh
+```
+
+Note that I had to edit the `setsas.sh` file to change the line near the end to `ulimit -s 65520`. The larger value for the stack size that is in the file by default didn't work because it was above the hard limit. I don't really understand this but the workaround seems to work (anyone know more about this than me!?)
```
Comment (fjebaker) [inline guides/sas-setup.md:30]: There is also `ulimit -s unlimited` which might be a bit more system agnostic, but doing so is generally not a great idea? Executables shouldn't really be bursting the stack, but SAS is what SAS is I suppose. I have no idea why that works either...

**Review by fjebaker — APPROVED** [review]

**Comment by fjebaker** [comment]
Is it worth adding a link to the [Docker SAS image](https://www.cosmos.esa.int/web/xmm-newton/sas-installation-docker4sas) somewhere in the guide too? Or maybe later we could add a more tutorial-esque setup for Docker, as you've done here for native :)

File: guides/sas-setup.md (hunk 1 above)
Comment (phajy) [reply guides/sas-setup.md:30]: I expect this problem will be resolved when SAS is updated in line with more recent versions of MacOS. Yes, not sure why a larger stack should be required...

File: guides/sas-setup.md [hunk 2]
```diff
@@ -28,6 +29,10 @@ source /Users/my_user/Software/xmm-sas/xmmsas_20211130_0941/setsas.sh
 
 Note that I had to edit the `setsas.sh` file to change the line near the end to `ulimit -s 65520`. The larger value for the stack size that is in the file by default didn't work because it was above the hard limit. I don't really understand this but the workaround seems to work (anyone know more about this than me!?)
 
+## Docker
+
+You can also run SAS using Docker, although at present (as of 2023-01-20) this is x86 only (but it does run OK on Mac with Apple Silicon). See the [Docker for SAS](https://www.cosmos.esa.int/web/xmm-newton/sas-installation-docker4sas) page for instructions.
+
```
Comment (phajy) [inline guides/sas-setup.md:35]: Added SAS Docker note as suggested

File: guides/xspec-setup.md [hunk 3]
```diff
@@ -117,6 +120,48 @@ docker run \
...
+Up-to-date as of 2023-01-20. See [Latest CALDB Files for Supported Missions](https://heasarc.gsfc.nasa.gov/docs/heasarc/caldb/caldb_supported_missions.html). Note that at the moment a patch file is required for NuSTAR.
+
+```
+cd $CALDB
+wget https://heasarc.gsfc.nasa.gov/FTP/caldb/data/nustar/fpm/goodfiles_nustar_fpm_20221229.tar.gz
+tar -zxvf goodfiles_nustar_fpm_20221229.tar.gz
+wget https://heasarc.gsfc.nasa.gov/FTP/caldb/data/nustar/fpm/goodfiles_nustar_fpm_clockcor_20221229.tar.gz
+tar -zxvf goodfiles_nustar_fpm_clockcor_20221229.tar.gz
+```
+
```
Comment (phajy) [inline guides/xspec-setup.md:164]: Also added notes about installing the calibration databases and files for specific missions.

**Review by fjebaker — APPROVED** [review]
This all looks great! Will merge 👍

**Pull request merged**